main-app/app/scripts/providers/generate-deck.js

```diff
@@ -0,0 +1,24 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.SnapCardGame.Deck')
+        .factory('GenerateDeck', ['ShuffleAndDeal', function (ShuffleAndDeal) {
+            var DeckGeneration = function () {
+                var deck = [];
+                this.makeDeck = function () {
+                    var suits = [1, 2, 3, 4],
```

Comment:
Suits doesn't buy you anything - you know you have 4 suits so you don't need this array.